### PR TITLE
chore(core): new networkrouter and type

### LINKED ISSUE #100: [Task](network):  BFF-Network create Schema for OpenStack API Network Response Using Zod
# Schema for type Network

We need to define a Zod schema to validate and type-check the response from the OpenStack Networks API. This schema will ensure that all responses conform to the expected structure, which helps in maintaining type safety throughout our application. By implementing this schema, we can facilitate better error handling and improve the robustness of our network operations within the OpenStack integration.

## Sub-tasks

- [ ] Analyze the current [OpenStack Networks API](https://docs.openstack.org/api-ref/network/v2/index.html) response structure and identify the required fields and their types.
  
- [ ] Define the Zod schema to represent the OpenStack Networks API response, including all necessary fields and validation rules.
  
- [ ] Implement tests using the defined Zod schema to ensure it correctly validates various response cases, including valid, invalid, and edge-case responses.

### PR DESCRIPTION
**Pull request by TilmanHaupt**
# Summary

This PR enhances the OpenStack API for Network by introducing comprehensive validation schemas for network-related data structures. It defines a base schema for network, extends it to include detailed network attributes, and structures the response schema for network lists. The implementation ensures type safety and data integrity throughout the API. Additionally, tests have been included to verify schema validation behavior.

Closes #100 

# Testing Instructions

1. `pnpm i`
2. `pnpm run test`

# Checklist

- [x] I have performed a self-review of my code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [ ] I have made corresponding changes to the documentation (if applicable).
- [x] My changes generate no new warnings or errors.

### CODE REVIEW

File: apps/aurora-portal/src/server/Network/routers/networkrouter.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { protectedProcedure } from "../../trpc"
+import { networkResponseSchema, Network } from "../types/models"
+
+export const networkRouter = {
+  getNetworks: protectedProcedure.query(async ({ ctx }): Promise<Network[] | undefined> => {
+    const openstackSession = ctx.openstack
```
Comment (andypf) [inline apps/aurora-portal/src/server/Network/routers/networkrouter.ts:6]: Please rename the function to getRoutersByProjectId, add input with zod validation. And please call `ctx.rescopeSession({projectId: input.projectId})`instead of `ctx.openstack`.

```ts
.input(z.object({projectId:z.string()}))
```

**Review by andypf — CHANGES_REQUESTED** [review]
Good Job. Please address my comments

**Review by andypf — APPROVED** [review]

**Review by barsukov — COMMENTED** [review]
Good work please one type maybe to add on example data wdyt?

**Review by barsukov — APPROVED** [review]
